packages/block-library/src/heading/autogenerate-anchors.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { deburr, trim } from 'lodash';
+
+/**
+ * Returns the text without markup.
+ *
+ * @param {string} text The text.
+ *
+ * @return {string} The text without markup.
+ */
+const getTextWithoutMarkup = ( text ) => {
+	const dummyElement = document.createElement( 'div' );
+	dummyElement.innerHTML = text;
+	return dummyElement.innerText;
+};
+
+/**
+ * Get the slug from the content.
+ *
+ * @param {string} content The block content.
+ *
+ * @return {string} Returns the slug.
+ */
+const getSlug = ( content ) => {
+	// Get the slug.
+	return trim(
+		deburr( getTextWithoutMarkup( content ) )
+			.replace( /[^\p{L}\p{N}]+/gu, '-' )
+			.toLowerCase(),
+		'-'
+	);
+};
+
+/**
+ * Generate the anchor for a heading.
+ *
+ * @param {string}   clientId          The block ID.
+ * @param {string}   content           The block content.
+ * @param {string[]} allHeadingAnchors An array containing all headings anchors.
+ *
+ * @return {string|null} Return the heading anchor.
+ */
+export const generateAnchor = ( clientId, content, allHeadingAnchors ) => {
+	const slug = getSlug( content );
+	// If slug is empty, then return null.
+	// Returning null instead of an empty string allows us to check again when the content changes.
+	if ( '' === slug ) {
+		return null;
+	}
+
+	delete allHeadingAnchors[ clientId ];
```

Comment:
Isn't it a set now? Set has several useful methods, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set.

Reply:
Changed `allHeadingAnchors` to an object so should be OK now :+1: